### PR TITLE
Integrate changes

### LINKED ISSUE #17: expose axis arguments as props

### LINKED ISSUE #24: don't bundle build

### PR DESCRIPTION
**Pull request by skeate**
I'm not entirely sure why we got so separated on this; we should've been committing changes to you the entire time.

Oh well.

This should include all the changes, and adjust the build so that all files are included (right now you're only building the index.js file and nothing else). It also does get rid of the dist folder being in the repo. Not sure why you added it; the commit message says to be able to install from github but why would you want to do that? Regardless I can add it back if you insist.

for issue management, this:

* closes #24 
* adds some things in #11 
* closes #17 (I think)

### CODE REVIEW

**Comment by skeate** [comment]
Removed the useless stylesheet, can't remember if there were other things

File: package.json [hunk 1]
```diff
@@ -52,9 +53,12 @@
     "webpack-dev-server": "^1.16.2",
     "webpack-validator": "^2.2.9"
   },
-  "dependencies": {
+  "peerDependencies": {
     "d3": "^4.3.0",
-    "react": "^0.14.7"
+    "react": "*"
```
Comment (jameslaneconkling) [inline package.json]: it's probably not the case that the lib can be this flexible.  Not sure if there would be a bottom limit to compatability.  Given that React continues to deprecate things, though, there's definitely an upper limit.

File: package.json [hunk 2]
```diff
@@ -52,9 +53,12 @@
...
   },
-  "main": "dist/index.js"
+  "main": "dist/index.js",
+  "dependencies": {
```
Comment (jameslaneconkling) [inline package.json]: remove

File: src/components/accessories/XAxis/XAxis.jsx [hunk 3]
```diff
@@ -20,10 +20,13 @@ export default class XAxis extends React.Component {
 
     xScale.rangeRound([0, containerWidth]);
 
-    $xAxis.call(
-      d3.axisBottom(xScale)
-        .tickPadding(10)
-    );
+    let axis = d3.axisBottom(xScale).ticks(d3.timeHour.every(3));
```
Comment (jameslaneconkling) [inline src/components/accessories/XAxis/XAxis.jsx:23]: ticks should be configurable, not hardcoded to timeHour.

Arguably, configuration should be standardized on [`tickArguments`](https://github.com/d3/d3-axis/blob/master/README.md#axis_tickArguments), which is a single interface that can configure both `ticks` and `tickFormat`.

For a shorter-term solution, though, it should just match YAxis.

File: src/svg/diagonalHatch.svg [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<svg>
```
Comment (jameslaneconkling) [inline src/svg/diagonalHatch.svg]: delete

File: webpack.config.js [hunk 5]
```diff
@@ -52,7 +52,11 @@ const loaders = PROD ?
       test: /\.scss$/,
       loader: 'style-loader!css-loader!sass-loader?sourceMap',
       include: path.join(__dirname, 'example')
-    }
+    },
+    // {
```
Comment (jameslaneconkling) [inline webpack.config.js]: delete

**Review by jameslaneconkling — COMMENTED** [review]
Looking at some of the new Chart components, it looks like some of them could be achieved by composing existing charts, rather than creating new ones.

E.g. `MultilineChartWithDots`: the multiline part could be achieved by composing multiple `LineChart`s; while the Dots part is essentially a scatterPlot.

The same _might_ apply for `GroupedBarChart`, `GroupedLayeredBarChart`, and maybe `FloatingBarChart` w/ `BarChart`, and `GroupedStackedBarChart` with `StackedBarChart`.  I'd have to look closer to make sure that is really the case, though.

File: package.json (hunk 1 above)
Comment (skeate) [reply package.json]: `>0.14`? I don't know that we should try to guess at an upper limit, and afaik it works with latest

**Comment by skeate** [comment]
bringing in @jedininjaster to comment

**Pull request closed without merging**